doc/nrf/drivers/uart_nrf_lpuart.rst

```diff
@@ -1,74 +1,98 @@
+#. The transmitter initiate the transmission by calling :cpp:func:`uart_tx`.
+#. The driver reconfigures the REQ line to input with pull up and enables the detection of high to low transition.
+   The line is set to high, due to the pull up register, and that triggers an interrupt on the RDY line.
+#. On that interrupt, the driver starts the UART receiver.
+#. When the receiver is ready, the driver reconfigures the RDY line to output low.
+#. Then, the driver reconfigures the RDY line to input with pull up and enables the interrupt on detection of high to low transition.
+#. This sequence results in a short pulse on the line, as the line goes low and high.
+#. The initiator detects the pulse and starts the standard UART transfer.
+#. When the transfer is completed, the driver reconfigure the REQ line to the initial state: output set to low.
+   This results in a line state change.
+#. As the line goes low, the receiver detects the change.
+   This indicates that the UART receiver can be stopped.
 
-Low Power UART configuration includes:
-:option:`CONFIG_NRF_LPUART_UARTE_INSTANCE`: UART instance choice
-:option:`CONFIG_NRF_LPUART_REQ_PIN`: Request line pin
-:option:`CONFIG_NRF_LPUART_RDY_PIN`: Ready line pin
+Once the receiver acknowledges the transfer, it must be capable of receiving the whole transfer until the REQ line goes down.
+
+This requirement can be fullfilled in two ways:
+
+* By providing a buffer of the size of the maximum packet length
+* By continuously responding to the RX buffer request event.
+  The latency of the event handling must be taken into account in that case.
+  For example, a flash page erase on some devices may have a significant impact.
+
+Configuration
+*************
+
+The low power UART driver is built on top of the standard UART driver extended by control lines protocol.
+It is configured in the device tree as the child node of the UART instance which extends standard UART configuration with REQ and RDY lines.
+Additionally, the standard UART configuration must have flow control and flow control pins disabled.
+
+See the following example:
+
+.. code-block:: DTS
+
+	&uart1 {
+		compatible = "nordic,nrf-uarte";
+		status = "okay";
+		rx-pin = <44>;
+		tx-pin = <45>;
+		baudrate = <1000000>;
+		/delete-property/ rts-pin;
+		/delete-property/ cts-pin;
+		/delete-property/ hw-flow-control;
+	
+		lpuart: nrf-sw-lpuart {
+			compatible = "nordic,nrf-sw-lpuart";
+			status = "okay";
+			label = "LPUART";
+			req-pin = <46>;
+			rdy-pin = <47>;
+		};
+	};
+
+The low power UART configuration includes:
+
+* :option:`CONFIG_NRF_LPUART_UARTE_INSTANCE`: the UART instance choice
+* :option:`CONFIG_NRF_LPUART_REQ_PIN`: the Request line pin
+* :option:`CONFIG_NRF_LPUART_RDY_PIN`: the Ready line pin
 
 Usage
 *****
 
-Low Power UART is accessed and control using Asynchronous UART API.
+You can access and control the low power UART using the Asynchronous UART API.
```

Comment:
We should add a link to this API. If I understand correctly, it's this one:
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/reference/peripherals/uart.html
@nordic-krch  please confirm
If yes, then it should be :ref:\`zephyr:uart_api\`

Reply:
waiting on this one for confirmation from @nordic-krch